File: lib/model1/contribuable1.dart
class Contribuable1 {
  final int id;
  final String taxPayerNo;
  final String nif;
  final String rs;
  final String? centre;
  final String? adresse;
  final String? phone;
  final String? email;
  final int? dernAnnee;
  final bool actif;
  final String? activite;

  Contribuable1({
    required this.id,
    required this.taxPayerNo,
    required this.nif,
    required this.rs,
    this.centre,
    this.adresse,
    this.phone,
    this.email,
    this.dernAnnee,
    required this.actif,
    this.activite,
  });

  factory Contribuable1.fromJson(Map<String, dynamic> json) {
    return Contribuable1(
      id: json['id'],
      taxPayerNo: json['tax_payer_no'],
      nif: json['nif'],
      rs: json['rs'],
      centre: json['centre'] ?? '',
      adresse: json['adresse'] ?? '',
      phone: json['phone'] ?? '',
      email: json['email'] ?? '',
      dernAnnee: json['dern_annee'],
      actif: json['actif'] ?? false,
      activite: json['activite'] ?? '',
    );
  }
}

File: backend/models1/Contribuable1.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Contribuable1 = sequelize.define('Contribuable1', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  tax_payer_no: { type: DataTypes.STRING(50), unique: true },
  nif: { type: DataTypes.STRING(50), unique: true },
  rs: { type: DataTypes.STRING(255) },
  centre: { type: DataTypes.STRING(100) },
  adresse: { type: DataTypes.STRING(255) },
  phone: { type: DataTypes.STRING(50) },
  email: { type: DataTypes.STRING(100) },
  dern_annee: { type: DataTypes.INTEGER },
  actif: { type: DataTypes.BOOLEAN, defaultValue: true },
  activite: { type: DataTypes.STRING(255) } // 🔹 nouveau champ
}, {
  tableName: 'contribuables1',
  timestamps: true,
});

module.exports = Contribuable1;
